src/spox/opset/ai/onnx/ml/v3.py

```diff
@@ -375,20 +360,6 @@ class Inputs(BaseInputs):
     class Outputs(BaseOutputs):
         Y: _VarInfo
 
-    def infer_output_types(self, input_prop_values: PropDict) -> dict[str, Type]:
-        if not self.inputs.fully_typed:
-            return {}
-        if self.attrs.cats_int64s:
-            n_encodings = len(self.attrs.cats_int64s.value)
-        elif self.attrs.cats_strings:
-            n_encodings = len(self.attrs.cats_strings.value)
-        else:
-            raise InferenceError(
-                "Either `cats_int64s` or `cats_strings` attributes must be set."
-            )
-        shape = (*self.inputs.X.unwrap_tensor().shape, n_encodings)  # type: ignore
-        return {"Y": Tensor(dtype=np.float32, shape=shape)}
-
```

Comment:
I believe the exact same comment as for CategoryMapper applies here, except that we already check for existence (just not length) upstream. After that we can remove this.